app/models/concerns/github_repoable.rb

```diff
@@ -8,14 +8,18 @@ def add_team_to_github_repository
     github_repository = GitHubRepository.new(organization.github_client, github_repo_id)
     github_team       = GitHubTeam.new(organization.github_client, github_team_id)
 
-    github_team.add_team_repository(github_repository.full_name)
+    options = {}
+    options[:permission] = 'admin' if give_admin_permission?
```

Comment:
👍 I like it. Do you prefer `assignment_permissions` over `repository_permissions`?